schema-layer/data-structures/vector.md

```diff
@@ -0,0 +1,361 @@
+         1. Create a copy of the current node, appending `value` to the end of the `data` array of the copy.
+         2. Set `mutated` to `true`
+         3. Repeat from step 3 with `value` set to the CID of the mutated node.
+      3. If the length of the `data` array at the current node is `width`, an overflow is necessary.
+         1. Create a new node with the same `height` as the current node and a `data` array that only contains the single `value`.
+         2. Repeat from step 3 with `value` set to the CID of the new node.
+4. If the tail chain is empty (this does not occur on the first pass as the root is always in the tail chain).
+   1. If `mutated` is still `false`, we have had cascading overflows up to the root so we need to add a new level to the Vector.
+      1. Create a new node with the original root node's `height + 1`, add two values to its `data` array: the CID of the original root node and `value` which will be the CID of a new, overflowed child node.
+      2. Return the CID of the new node, which becomes the new root node.
+   2. If `mutated` is `true`
+      1. Return `value`, which should be the CID of the last mutated node, which becomes the new root node.
+
+### `Size()`
+
+The `Size` algorithm uses a subtractive process, first calculating the maximum potential size and then iterating through the tail chain to subtract known empty portions given varying `data` array sizes at each `height`. If `data` arrays are maximally full at each `height` then the original maximum potential size calculation is used as the correct value.
+
+1. Set `size` to be the maximum potential size of the data structure with `width`<sup>`height + 1`</sup>.
+2. Calculate the number of empty potential value slots at this height with `width`<sup>`height * (width - dataLength)`</sup>, where `dataLength` is the number of elements in the current node's `data` array and `height` is the current node's `height`. Subtract this value from `size` and set the result as the new `size`.
+3. If the current node's `height` is `0`.
+   1. Return `size` as the correct size.
+4. If the current node's `height` is greater than `0`.
+   1. Locate the next tail node identified by the CID at the last position of the current node's `data` array.
+   2. Repeat from step 2 with the next tail node as the "current node".
+
+### `CreateFrom(array, width)`
+
+1. Set `height` to `0`, indicating that each node created at this `height` will have leaf values in it rather than links to child nodes.
+2. Calculate `nodeCount` for this `height` with `ceiling(arrayLength / width)`, where `arrayLength` is the number of elements in `array`.
+3. Create a temporary `nextArray` array which starts empty but will fill up to `nodeCount`.
+4. Iterate with an incrementing integer `i` from `0` up to, but not including `nodeCount`.
+   1. Take a slice of `array` beginning at element `i x width` and ending at the element before `(i + 1) x width` or `arrayLength`, whichever is least.
```

Comment:
In other parts you've used `*` as symbol for multiplications.

Reply:
I meant to use `x`, I corrected the previous use of `*`